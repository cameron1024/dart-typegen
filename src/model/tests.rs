use super::*;

macro_rules! parse_snapshot {
    ($name:ident) => {
        #[test]
        fn $name() {
            let text = include_str!(crate::test_file!($name));
            let config = Library::parse_impl(None, text).unwrap();

            insta::assert_debug_snapshot!(config);
        }
    };
}

crate::all_test_files!(parse_snapshot);
