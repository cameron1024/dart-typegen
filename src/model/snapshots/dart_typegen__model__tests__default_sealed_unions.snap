---
source: src/model/tests.rs
expression: ron
---
Library {
    preamble: None,
    postamble: None,
    defaults: Some(
        Defaults {
            union: Some(
                UnionOptions {
                    sealed: Some(
                        SpannedScalar {
                            value: true,
                            span: SourceSpan {
                                offset: SourceOffset(
                                    75,
                                ),
                                length: 4,
                            },
                        },
                    ),
                    json_discriminant: None,
                },
            ),
            field: None,
        },
    ),
    enums: [],
    classes: [],
    unions: [
        Union {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Foo",
                span: SourceSpan {
                    offset: SourceOffset(
                        93,
                    ),
                    length: 5,
                },
            },
            sealed: None,
            json_discriminant: None,
            docs: None,
            classes: [
                Class {
                    span: Span(
                        0,
                        0,
                    ),
                    name: SpannedScalar {
                        value: "X",
                        span: SourceSpan {
                            offset: SourceOffset(
                                109,
                            ),
                            length: 3,
                        },
                    },
                    fields: [],
                    docs: None,
                    extra_dart: [],
                },
                Class {
                    span: Span(
                        0,
                        0,
                    ),
                    name: SpannedScalar {
                        value: "Y",
                        span: SourceSpan {
                            offset: SourceOffset(
                                121,
                            ),
                            length: 3,
                        },
                    },
                    fields: [],
                    docs: None,
                    extra_dart: [],
                },
            ],
            extra_dart: [],
        },
    ],
}
